base_layer/core/src/validation/transaction_validators.rs

```diff
@@ -71,6 +71,10 @@ impl<B: BlockchainBackend> MempoolTransactionValidation for TxInternalConsistenc
             return Err(ValidationError::ErroneousCoinbaseOutput);
         }
 
+        // We can call this function with a constant value, because we've just shown that this is NOT a coinbase, and
+        // only coinbases may have the extra field set (the only field that the fn argument affects).
+        tx.body.check_output_features(1)?;
```

Comment:
Not the cleanest check, tbh, But getting the actual max size into this method would mean adding `rules` as an arg to a whole bunch of methods